core/validation/failures/user.py

```diff
@@ -312,13 +313,12 @@ def failures_to_messages(
     # group cells by sheet, section and desc
     error_messages = group_validation_messages(error_messages)
     error_messages.sort()
-
     validation_errors = [
```

Comment:
Could you put the error_type as an attribute of the validation error. This groups them and should remove the need for your submit PR

Reply:
I can see two ways to do this:
- Add error type as a return value from ever validation error and change all tests to reflect this - Lots of work
- Add error types into the validation errors here - Quick, but a bit messy
```
for validation_error, error_type in zip(validation_errors,error_types):
    validation_error["type"] = error_type
```

Reply:
The difficulty with this is that some errors are combined in a previous step so they won't always line up to be zipped like this